src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -17,6 +24,13 @@ public Machine save(Machine machine) {
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Machine> query = criteriaBuilder.createQuery(Machine.class);
+            Root<Machine> root = query.from(Machine.class);
+            query.where(criteriaBuilder.greaterThan(root.get("year"), age));
```

Comment:
I guess the year for the car is the year it was made in, and age will be current_year - year